### PR TITLE
TF BART models - Add `cross_attentions` to model output and fix cross-attention head masking

### LINKED ISSUE #10698: Add `cross_attentions` to the output of TensorFlow encoder-decoder models
# 🚀 Feature request

TensorFlow encoder-decoder models cannot return `cross_attentions` as do their PyTorch counterparts.

## Motivation
It would be nice to narrow the gap between PyTorch and Tensorflow implementations.

## Your contribution
I've been working on PR fixing this issue.

## Reviewers
@jplu and whoever else within the community

### PR DESCRIPTION
**Pull request by stancld**
This PR fixes some missing and invalid things around `cross_attentions` for the TensorFlow implementation of BART models:

- `Bart`,
- `Blenderbot` / `Blenderbot_small`,
- `Marian`,
- `MBart`,
- `Pegasus`.

More specifically, this PR includes:
- Enable returning `cross_attentions`
- Add class `TFBaseModelOutputWithCrossAttentions` (according to the PyTorch counterpart) to support output containing `cross_attentions`
- Fix attention head masking for the cross-attention module (by the introduction of `cross_attn_head_mask` and `cross_attn_layer_head_mask`)
- Implement `test_head_masking` for `cross_attn_head_mask`
- Fix some little typos in docs
- Update model templates - implement `head_mask`, `decoder_head_mask`, `cross_attn_head_mask` and code around `cross_attentions` to the TF encoder-decoder models

<hr>
Partially fixes: #10698 

<hr>

**Reviewers:** @jplu @patrickvonplaten @LysandreJik @sgugger

### CODE REVIEW

**Comment by stancld** [comment]
Currently, there are some Torch/TF equivalent tests failing but it should be settled down once #10605 be merged.

**Review by jplu — COMMENTED** [review]
Thanks a lot for this nice PR!

Did you run all the slow tests as well on the impacted models? If not please can you do it, in order to be sure that everything still works as expected :)

**Comment by stancld** [comment]
@jplu - I'm gonna run all slow tests today and will let you know if everything works or not.

**Comment by stancld** [comment]
@jplu - I ran all (slow) non-GPU tests and it seems to me everything is passing :)

**Comment by jplu** [comment]
Ok, if all the tests for the involved models, including the slow ones, are passing, it is fine to merge for me.

**Comment by patrickvonplaten** [comment]
still open

**Comment by stancld** [comment]
@jplu @patrickvonplaten After merging #10605 and rebasing the branch, all tests have passed now :)

**Comment by patrickvonplaten** [comment]
Great, thanks @stancld! 

This looks good to merge for me.

**Review by patrickvonplaten — APPROVED** [review]

**Comment by patrickvonplaten** [comment]
Pinging @Rocketknight1 for a final review here. Seq2Seq models like BART should also return cross attentions in TF (not just in PT)

**Comment by Rocketknight1** [comment]
This is a big PR but LGTM! I haven't exhaustively checked everything but the changes seem correct and innocuous, so if it passes tests I'm happy to merge it.

**Review by Rocketknight1 — APPROVED** [review]
